test_plugin/src/lib.rs

```diff
@@ -51,3 +51,27 @@ pub fn op_test_async(data: &[u8], zero_copy: Option<PinnedBuf>) -> CoreOp {
 
   Op::Async(fut.boxed())
 }
+
+struct TestResource {
+  pub name: String,
+}
+
+impl Drop for TestResource {
+  fn drop(&mut self) {
+    println!("Dropped resource: {}", self.name)
+  }
+}
+
+impl Resource for TestResource {}
+
+pub fn op_create_resource(
+  state: &PluginState,
+  data: &[u8],
+  _zero_copy: Option<PinnedBuf>,
+) -> CoreOp {
+  let name = std::str::from_utf8(&data[..]).unwrap().to_string();
+  let mut table = state.lock_resource_table();
+  let resource = TestResource { name };
+  let rid = table.add("testResource", Box::new(resource));
+  Op::Sync(Box::new(rid.to_be_bytes()))
+}
```

Comment:
Creating resources is certainly a desirable thing to be able to do, but I'm bothered that internal ops access the resource table differently than plugins...

Reply:
> I'm bothered that internal ops access the resource table differently than plugins...

I'm not sure what differences you are referring to here.